.github/workflows/testing.yaml

```diff
@@ -133,3 +133,52 @@ jobs:
 
       - name: Package
         run: poetry build
+
+  dependency-matrix-rest:
+    # The dependency matrix is used to test the compatibility of the 
+    # SDK with different dependency versions customers might use.
+    # The test itself is a single integration test that creates an index,
+    # adds a few vectors, and runs one query.
+    name: Dep matrix
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 5
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.8
+          - 3.9
+          - "3.10"
+          - 3.11
+        urllib3-version:
+          - 1.26.16
+          - 1.26.18
+          - 2.0.2
+          - 2.0.5
+          - 2.1.0
+          # - 1.26.17
+          # - 2.0.3
+          # - 2.0.4
+          # - 2.0.6
+          # - 2.0.7
```

Comment:
Can these comments be removed or is the intention to add them back in later?

Reply:
I haven't decided yet. Testing every single release kinda seems like overkill. The max jobs per workflow is 256 so I think I should keep it to this abbreviated list for now and save space for matrix runs on other jobs in the workflow.